astropy/coordinates/representation.py

```diff
@@ -538,12 +773,24 @@ def _combine_operation(self, op, other, reverse=False):
             Whether the operands should be reversed (e.g., as we got here via
             ``self.__rsub__`` because ``self`` is a subclass of ``other``).
         """
+        self._raise_if_has_differentials(op.__name__)
+
         result = self.to_cartesian()._combine_operation(op, other, reverse)
         if result is NotImplemented:
             return NotImplemented
         else:
             return self.from_cartesian(result)
 
+    # We need to override this setter to support differentials
+    @BaseRepresentationOrDifferential.shape.setter
+    def shape(self, shape):
+        # See: https://stackoverflow.com/questions/3336767/ for an example
+        BaseRepresentationOrDifferential.shape.fset(self, shape)
```

Comment:
Not clear on what you're suggesting - right now, the differentials added to a representation must have the same shape as the representation, so if we just keep that strictness for now we shouldn't have to deal with this?